### PR TITLE
Layer visibility

### PR DESCRIPTION
**Pull request by TheCovenant**

### CODE REVIEW

File: app/geo/gui/src/ts/twxplore/gui/geo/components/selectionPanel/SelectionPanel.tsx [hunk 1]
```diff
@@ -33,16 +33,34 @@ const SelectionPanelImpl: React.FunctionComponent = () => {
...
     (rootState: RootState) => rootState.app.map
   );
+
+  const keplerState: any = useSelector(
+    (rootState: RootState) => rootState.keplerGl
+  );
   const dispatch = useDispatch();
 
-  const handleChange = (typeEnum: FeatureType) => (
+  const getLayerIndex = (
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/selectionPanel/SelectionPanel.tsx:42]: You're only using this code in one place, why is it a function?

File: app/geo/gui/src/ts/twxplore/gui/geo/components/selectionPanel/SelectionPanel.tsx [hunk 2]
```diff
@@ -33,16 +33,34 @@ const SelectionPanelImpl: React.FunctionComponent = () => {
...
+    const newLayerConfig = {
+      isVisible: !state.featuresByType[featureType].visible,
+    };
+    dispatch(layerConfigChange(keplerLayers[layerIndex], newLayerConfig));
+    dispatch(changeTypeVisibility(featureType));
   };
 
-  const logger : Logger = React.useContext(LoggerContext)
-  logger.info("test")
+  const logger: Logger = React.useContext(LoggerContext);
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/components/selectionPanel/SelectionPanel.tsx:62]: Nice to see this.

File: app/geo/gui/src/ts/twxplore/gui/geo/reducers/map/mapReducer.ts [hunk 3]
```diff
@@ -62,6 +61,15 @@ export const mapReducer = (state: MapState, action: BaseAction): MapState => {
         result.featuresByType[
           feature.type! as keyof typeof result.featuresByType
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/reducers/map/mapReducer.ts:62]: I see the same expression (result.featuresByType[...]) over and over again, please put into a variable.

File: app/geo/gui/src/ts/twxplore/gui/geo/states/map/initialMapState.ts [hunk 4]
```diff
@@ -29,6 +21,7 @@ Object.values(FeatureType).map(featureType => {
     dirty: false,
     featureTypeState: MapFeatureTypeState.ABSENT_ON_MAP,
     attributeStates: {},
+    visible: null,
```
Comment (gordom6) [inline app/geo/gui/src/ts/twxplore/gui/geo/states/map/initialMapState.ts:24]: Why do you need null? Why not default to false?

Comment (TheCovenant) [reply app/geo/gui/src/ts/twxplore/gui/geo/states/map/initialMapState.ts:24]: Null was to indicate absence from the map completely (i.e. a feature of that type has yet to be added/loaded).
Changed by defaulting to false and using MapFeatureTypeState.ABSENT_ON_MAP instead.

**Pull request merged**